python/paddle/fluid/contrib/slim/tests/CMakeLists.txt

```diff
@@ -1,15 +1,57 @@
 file(GLOB TEST_OPS RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" "test_*.py")
 string(REPLACE ".py" "" TEST_OPS "${TEST_OPS}")
 
+function(inference_analysis_python_api_int8_test target model_dir data_dir filename)
+    py_test(${target} SRCS ${filename}
+        ENVS FLAGS_OMP_NUM_THREADS=${CPU_NUM_THREADS_ON_CI}
+        ARGS --infer_model ${model_dir}/model
+             --infer_data ${data_dir}/data.bin
+             --int8_model_save_path int8_models/${target} 
+             --warmup_batch_size 100
+             --batch_size 50)
+endfunction()
+
 # NOTE: TODOOOOOOOOOOO
 # temporarily disable test_distillation_strategy since it always failed on a specified machine with 4 GPUs
 # Need to figure out the root cause and then add it back
 list(REMOVE_ITEM TEST_OPS test_distillation_strategy)
 
-if(APPLE OR WIN32 OR NOT WITH_MKL)
-    list(REMOVE_ITEM TEST_OPS test_mkldnn_int8_quantization_strategy)
+# int8 image classification python api test
+if(LINUX AND WITH_MKLDNN)
+  set(INT8_DATA_DIR "${INFERENCE_DEMO_INSTALL_DIR}/int8v2")
+
+  #resnet50 int8
+  set(INT8_RESNET50_MODEL_DIR "${INT8_DATA_DIR}/resnet50")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_resnet50 ${INT8_RESNET50_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
+
+  #mobilenet int8
+  set(INT8_MOBILENET_MODEL_DIR "${INT8_DATA_DIR}/mobilenet")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_mobilenet ${INT8_MOBILENET_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
+
+  #mobilenetv2 int8
+  set(INT8_MOBILENETV2_MODEL_DIR "${INT8_DATA_DIR}/mobilenetv2")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_mobilenetv2 ${INT8_MOBILENETV2_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
+
+  #resnet101 int8
+  set(INT8_RESNET101_MODEL_DIR "${INT8_DATA_DIR}/resnet101")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_resnet101 ${INT8_RESNET101_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
+
+  #vgg16 int8
+  set(INT8_VGG16_MODEL_DIR "${INT8_DATA_DIR}/vgg16")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_vgg16 ${INT8_VGG16_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
+
+  #vgg19 int8
+  set(INT8_VGG19_MODEL_DIR "${INT8_DATA_DIR}/vgg19")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_vgg19 ${INT8_VGG19_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
+
+  #googlenet int8
+  set(INT8_GOOGLENET_MODEL_DIR "${INT8_DATA_DIR}/googlenet")
+  inference_analysis_python_api_int8_test(test_analyzer_int8_googlenet ${INT8_GOOGLENET_MODEL_DIR} ${INT8_DATA_DIR} test_mkldnn_int8_quantization_strategy.py)
 endif()
 
+list(REMOVE_ITEM TEST_OPS test_mkldnn_int8_quantization_strategy)
```

Comment:
@wzzju how about this version of cmake

Reply:
how about https://github.com/PaddlePaddle/Paddle/pull/17634#discussion_r290310558?